controllers/nodeagent.go

```diff
@@ -306,58 +366,58 @@ func (r *DPAReconciler) customizeResticDaemonset(dpa *oadpv1alpha1.DataProtectio
 	return ds, nil
 }
 
-func (r *DPAReconciler) ReconcileResticRestoreHelperConfig(log logr.Logger) (bool, error) {
+func (r *DPAReconciler) ReconcileFsRestoreHelperConfig(log logr.Logger) (bool, error) {
 	dpa := oadpv1alpha1.DataProtectionApplication{}
 	if err := r.Get(r.Context, r.NamespacedName, &dpa); err != nil {
 		return false, err
 	}
 
-	resticRestoreHelperCM := corev1.ConfigMap{
+	fsRestoreHelperCM := corev1.ConfigMap{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      ResticRestoreHelperCM,
+			Name:      FsRestoreHelperCM,
 			Namespace: r.NamespacedName.Namespace,
 		},
 	}
 
-	op, err := controllerutil.CreateOrPatch(r.Context, r.Client, &resticRestoreHelperCM, func() error {
+	op, err := controllerutil.CreateOrPatch(r.Context, r.Client, &fsRestoreHelperCM, func() error {
```

Comment:
Is it safe to remove the resticRestoreHelperCM ?

We have other options:
 1. Error (so user will know what's wrong)
 2. Re-label and give some warning, add some prefix to the labels so the user won't loose it
 3. Rename to fsRestoreHelper if fsRestoreHelper does not exists, otherwise option (1) or (2)
 4. Delete resticRestoreHelperCM if fsRestoreHelper is created or patched and have some warning

Reply:
I am going for an implementation that we talked over with @shawn-hurley. @sseago please let me know if that should be modified:

1. Check if the ConfigMap `restic-restore-action-config` exists.
2. If it exists check if it was created by OADP (by comparing the ConfigMap values to the one OADP had).
3a. If the ConfigMap was unmodified, we can delete it and create `fs-restore-action-config` (with of course event&warning on the DPA)
3b. If the ConfigMap `restic-restore-action-config` was modified, it means we should not delete it. Instead modify the `restic-restore-action-config` CM by adding some `-deprecated` suffix to the label, so velero won't find it and migrate this ConfigMap to the `fs-restore-action-config` with exactly same values (again event & warning on the DPA).

That will also modify a bit our design doc, so users know what to expect and also we deprecate that CM in the 1.4.

Reply:
@mpryc We shouldn't error, since the old configmap *will* exist on upgrade. It's not user-configurable, OADP created it, and OADP should remove it on upgrade. Currently this configmap is not user-configurable -- oadp sets it, so it should be sufficient to simply remove the old one and create the new one.

Reply:
Currently if user modifies the configmap, the modifications are completely overwritten by the operator. We may need to make this configurable in a future enhancement, but we don't need to worry about custom changes for now, since they won't exist.

Reply:
Ok removed in the last revision.